filecoin-proofs/src/api.rs

```diff
@@ -67,38 +67,230 @@ pub struct SealOutput {
     pub piece_inclusion_proofs: Vec<PieceInclusionProof<PedersenHasher>>,
 }
 
-/// Generates a proof-of-spacetime, returning and detected storage faults.
+/// The minimal information required about a replica, in order to be able to generate
+/// a PoSt over it.
+#[derive(Clone, Debug, PartialEq, Eq, Hash)]
+pub struct ReplicaInfo {
+    /// Path to the replica.
+    access: String,
+    /// The replica commitment.
+    commitment: Commitment,
+}
+
+impl std::cmp::Ord for ReplicaInfo {
+    fn cmp(&self, other: &Self) -> std::cmp::Ordering {
+        self.commitment.as_ref().cmp(other.commitment.as_ref())
+    }
+}
+
+impl std::cmp::PartialOrd for ReplicaInfo {
+    fn partial_cmp(&self, other: &Self) -> Option<std::cmp::Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+impl ReplicaInfo {
+    pub fn new(access: String, commitment: Commitment) -> Self {
+        ReplicaInfo { access, commitment }
+    }
+
+    pub fn safe_commitment(&self) -> Result<PedersenDomain, failure::Error> {
+        as_safe_commitment(&self.commitment, "comm_r")
+    }
+
+    /// Generate the merkle tree of this particular replica.
+    pub fn merkle_tree(&self, sector_size: u64) -> Result<Tree, Error> {
+        let mut f_in = File::open(&self.access)?;
+        let mut data = Vec::new();
+        f_in.read_to_end(&mut data)?;
+
+        let bytes = PaddedBytesAmount(sector_size as u64);
+        public_params(bytes, 1).graph.merkle_tree(&data)
+    }
+}
+
+fn as_safe_commitment(
+    comm: &Commitment,
+    commitment_name: impl AsRef<str>,
+) -> Result<PedersenDomain, failure::Error> {
+    bytes_into_fr::<Bls12>(comm).map(Into::into).map_err(|err| {
+        format_err!(
+            "Invalid commitment ({}): {:?}",
+            commitment_name.as_ref(),
+            err
+        )
+    })
+}
+
+/// Generates a proof-of-spacetime.
 /// Accepts as input a challenge seed, configuration struct, and a vector of
-/// sealed sector file-path plus CommR tuples.
-///
+/// sealed sector file-path plus CommR tuples, as well as a list of faults.
```

Comment:
> as well as a list of faults.

This comment tells lies!

Reply:
It does not, this function takes a list of faults as inputs. `faults: &[u64]`

Reply:
I tell lies! The comment tells truths.